src/View/SSTemplateParser.peg

```diff
@@ -274,8 +265,8 @@ class SSTemplateParser extends Parser implements TemplateParser
     }
 
     /**
-     * The basic generated PHP of LookupStep and LastLookupStep is the same, except that LookupStep calls 'obj' to
-     * get the next ModelData in the sequence, and LastLookupStep calls different methods (XML_val, hasValue, obj)
+     * The basic generated PHP of LookupStep and LastLookupStep is the same, except that LookupStep calls 'scopeToIntermediateValue' to
+     * get the next ModelData in the sequence, and LastLookupStep calls different methods (getOutputValue, hasValue, scopeToIntermediateValue)
```

Comment:
As you can see here, `$scope->obj()` has been replaced with `$scope->scopeToIntermediateValue()` (i.e. change the current scope to THIS thing), and `$scope->XML_val()` (which was handled by the magic `SSViewer_Scope::__call()` before) is now `$scope->getOutputValue()` (i.e. this is the specific value that will be directly output into the rendered result).

These new method names are a lot more descriptive, aren't likely to cause confusion with similarly named methods elsewhere (you might notice those old methods are names of methods on `ModelData` - but didn't necessarily call them directly), and allow concrete methods to be implemented on `SSViewer_Scope` isntead of relying heaving on the magic of `__call()`